File: LeetCode/Easy/0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
class MyStack:

    def __init__(self):
        self.last = 0
        self.q = []

    def push(self, x: int) -> None:
        self.q.append(x)
        self.last += 1

    def pop(self) -> int:
        # self.q.pop()
        # self.last = self.q[len(self.q)-1]
        # len(self.q)-1
        if self.last == 0:
            return
        self.last -= 1
        return self.q.pop()

    def top(self) -> int:
        return self.q[self.last-1]

    def empty(self) -> bool:
        # print(self.last)
        if self.last:
            return False
        return True


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: LeetCode/Easy/0225-implement-stack-using-queues/0225-implement-stack-using-queues.cpp
class MyStack {
public:
    MyStack() {
        // global int last = 0;

    }
    
    void push(int x) {
        q.push_back(x);
    }
    
    int pop() {
        int last = q.back();
        q.pop_back();
        return last;
    }
    
    int top() {
        return q.back();
    }
    
    bool empty() {
        if(q.size())
            return false;
        return true;
    }
private:
    vector<int> q;
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */